sway-lsp/src/core/document.rs

```diff
@@ -162,7 +162,7 @@ impl TextDocument {
             pkg::ManifestFile::from_dir(&manifest_dir, forc::utils::SWAY_GIT_TAG).unwrap();
         let lock_path = forc_util::lock_path(manifest.dir());
         let plan = pkg::BuildPlan::from_lock_file(&lock_path, forc::utils::SWAY_GIT_TAG).unwrap();
```

Comment:
Ahh I wonder if this `BuildPlan` should be constructed via `load_from_manifest` rather than `from_lock_file` - otherwise LSP won't pick up changes to the manifest until the user manually triggers a build and updates the lock file again. @JoshuaBatty does this sound right?

If we can change this over to `load_from_manifest`, we should be able to make `from_lock_file` private.

Reply:
Depending on the input from @JoshuaBatty maybe we can do something like 

```rust
let plan =
  pkg::BuildPlan::load_from_manifest(&manifest, false, false, forc::utils::SWAY_GIT_TAG)
      .unwrap_or_else(|_| {
          pkg::BuildPlan::from_lock_file(&lock_path, forc::utils::SWAY_GIT_TAG)
              .map(|plan| plan.0)
              .unwrap()
      });
```

Reply:
I think rather than falling back to `from_lock_file`, we might be better off producing an error from LSP to the user so that they're aware of whatever the failure is. It might be a bit unexpected if LSP keeps using an old `BuildPlan` after they've updated the manifest.

Reply:
If I add a create to the `Cargo.toml` file and save it in a rust project then I am able to start using types from that crate and even jump to the definition of items in that crate without needing to build the project again. Not sure if it's rust or rust-analyzer that is downloading the added crate once the manifest file has been saved. It would be nice if we could replicate this behavior somehow.

Reply:
In that case it sounds like we do want to change the use of `from_lock_file` to `load_from_manifest here - @kayagokalp are you happy to do so in this PR so we can make `from_lock_file` private too?

Reply:
Should be good to go but didn't add an explicit error throwing for the failure as we didn't have one for the current version too:

https://github.com/FuelLabs/sway/blob/92ec0a655bd11d19b5595fcaa940e233feee883a/sway-lsp/src/core/document.rs#L164

Should we change the current behavior and explicitly return an error? or keep the current logic there (which is simply unwrapping)